### PR TITLE
feat(query): adjust downsample query

### PR DESCRIPTION
**Pull request by shinyano**
# what's wrong
The key column in previous downsample query result is confusing. It refers to the start of a window in such results, while user would assume it is a key for a record.

Also, user should be able to slice windows based on queried data, instead of specifying key range.

# fix
- Add two columns: `window_start` and `window_end` to avoid misunderstanding.
- Adjust downsample sql statement

## SQL adjust
Allow user to use:
```sql
SELECT <expression> (, <expression>)* FROM prefixPath <whereClause>? <downsamplingClause>?
<downsamplingClause>: OVER WINDOW LEFT_BRACKET SIZE DURATION (IN <timeInterval>)? (SLIDE DURATION)? RIGHT_BRACKET;
```
to generate windows according to queried data.

Example:
```sql
insert into test(key,a) values(3,4),(4,5),(5,6),(6,7),(7,8);
select a from test;
/*
ResultSets:
+---+------+
|key|test.a|
+---+------+
|  3|     4|
|  4|     5|
|  5|     6|
|  6|     7|
|  7|     8|
+---+------+
Total line number = 5
*/

select max(a) from test over window (size 2);
/*
ResultSets:
+---+------------+----------+-----------+
|key|window_start|window_end|max(test.a)|
+---+------------+----------+-----------+
|  3|           3|         4|          5|
|  5|           5|         6|          7|
|  7|           7|         8|          8|
+---+------------+----------+-----------+
Total line number = 3
*/

select max(a) from test over window (size 2 in [0,10]);
/*
ResultSets:
+---+------------+----------+-----------+
|key|window_start|window_end|max(test.a)|
+---+------------+----------+-----------+
|  2|           2|         3|          4|
|  4|           4|         5|          6|
|  6|           6|         7|          8|
+---+------------+----------+-----------+
Total line number = 3
*/
```

doc:https://oxlh5mrwi0.feishu.cn/wiki/BUerw0yvQiGBgOkz6YjcfbXInvd

### CODE REVIEW

File: thrift/src/main/proto/rpc.thrift [hunk 1]
```diff
@@ -321,6 +321,7 @@ struct DownsampleQueryReq {
     7: optional list<map<string, list<string>>> tagsList
     8: optional TimePrecision timePrecision
     9: optional TagFilterType filterType
+    10: required bool setInterval
```
Comment (SolomonAnn) [inline thrift/src/main/proto/rpc.thrift:324]: 变量最好不要以set开头，因为setter方法会变成setSetXXX

**Review by SolomonAnn — APPROVED** [review]

**Review by zhuyuqing — CHANGES_REQUESTED** [review]
需要去掉window_start, window_end两列

**Comment by shinyano** [comment]
> 需要去掉window_start, window_end两列

flink 滑动窗口文档
https://nightlies.apache.org/flink/flink-docs-master/zh/docs/dev/table/sql/queries/window-tvf/#%e6%bb%91%e5%8a%a8%e7%aa%97%e5%8f%a3hop

**Comment by zhuyuqing** [comment]
那调整一下语法，调整后的语法在会上过一下

**Pull request merged**